### PR TITLE
Build CUDA and DML together

### PR DESCRIPTION
**Pull request by mszhanyi**
### Description
Now, we need to build cuda and dml in one package.
But CUDA EP and DML EP can't run in one process.
It will throw the exception of `the GPU device instance has been suspended`
So the issue is CUDA EP and DML EP coexist in compile time but can't exist in run time.

This PR is to split cuda ep test and dml ep test in all unit tests.
The solution is to use 2 environment variable, NO_CUDA_TEST and NO_DML_TEST, in CI.

For example, if NO_CUDA_TEST is set, the DefaultCudaExecutionProvider will be nullptr, and the test will not run with CUDA EP.
In debugging, the CUDAExecutionProvider will not be called. 
I think, as long as cuda functions, like cudaSetDevice, are not called, DML EP tests can pass.

Disabled java test of testDIrectML because it doesn't work now even without CUDA EP.

### CODE REVIEW

**Comment by snnn** [comment]
Related #22597

File: java/src/test/java/ai/onnxruntime/InferenceTest.java [hunk 1]
```diff
@@ -740,7 +740,14 @@
   @Test
   @EnabledIfSystemProperty(named = "USE_DML", matches = "1")
   public void testDirectML() throws OrtException {
-    runProvider(OrtProvider.DIRECT_ML);
+    if (System.getProperty("USE_CUDA") != "1") {
```
[the remark this answers is not in the document]
Comment (mszhanyi) [reply java/src/test/java/ai/onnxruntime/InferenceTest.java]: fixed!

File: java/src/test/java/ai/onnxruntime/InferenceTest.java [hunk 2]
```diff
@@ -740,7 +740,14 @@
...
+      runProvider(OrtProvider.DIRECT_ML);
+    } else if(System.getProperty("USE_CUDA") == "1" && System.getenv("NO_CUDA_TEST") == "1" ) {
```
[the remark this answers is not in the document]
Comment (mszhanyi) [reply java/src/test/java/ai/onnxruntime/InferenceTest.java]: fixed!

[the remark this answers is not in the document]
Comment (mszhanyi) [reply java/src/test/java/ai/onnxruntime/InferenceTest.java]: fixed!

**Review by snnn — APPROVED** [review]

**Pull request merged**